### PR TITLE
[ZEPPELIN-5663] Provide REST API about notebook version control

### PR DESCRIPTION
**Pull request by huage1994**
### What is this PR for?
This PR is to provide REST API about notebook version control

### What type of PR is it?
[Feature]

### Todos
* [ ] - Task

### What is the Jira issue?
* Open an issue on Jira https://issues.apache.org/jira/browse/ZEPPELIN-5663

### How should this be tested?
* CI passed

### Questions:
* Does the licenses files need update? No
* Is there breaking changes for older versions? No
* Does this needs documentation? Yes

### CODE REVIEW

File: zeppelin-server/src/main/java/org/apache/zeppelin/service/NotebookService.java [hunk 1]
```diff
@@ -1012,12 +1012,12 @@ public void setNoteRevision(String noteId,
 
   }
 
-  public void getNotebyRevision(String noteId,
+  public Note getNotebyRevision(String noteId,
```
Comment (Reamer) [inline zeppelin-server/src/main/java/org/apache/zeppelin/service/NotebookService.java:1015]: Could you please put the following comment to this method.
`notebook.getNoteByRevision(...) does not use the NoteCache, so we can return a Note object here.`

File: zeppelin-server/src/test/java/org/apache/zeppelin/rest/NotebookRestApiTest.java [hunk 2]
```diff
@@ -164,6 +165,112 @@ public void testGetNoteByPathWithPathNotExist() throws IOException {
...
   }
 
+  @Test
+  public void testGetNoteRevisionHistory() throws IOException {
+    LOG.info("Running testGetNoteRevisionHistory");
+    String note1Id = null;
+    Notebook notebook = TestUtils.getInstance(Notebook.class);
+    try {
+      String notePath = "note1";
+      note1Id = TestUtils.getInstance(Notebook.class).createNote(notePath, anonymous);
```
Comment (Reamer) [inline zeppelin-server/src/test/java/org/apache/zeppelin/rest/NotebookRestApiTest.java:175]: Use `notebook` here.

File: zeppelin-server/src/test/java/org/apache/zeppelin/rest/NotebookRestApiTest.java [hunk 3]
```diff
@@ -164,6 +165,112 @@ public void testGetNoteByPathWithPathNotExist() throws IOException {
...
+  }
+
+  @Test
+  public void testGetNoteByRevision() throws IOException {
+    LOG.info("Running testGetNoteByRevision");
+    String note1Id = null;
+    Notebook notebook = TestUtils.getInstance(Notebook.class);
+    try {
+      String notePath = "note1";
+      note1Id = TestUtils.getInstance(Notebook.class).createNote(notePath, anonymous);
```
Comment (Reamer) [inline zeppelin-server/src/test/java/org/apache/zeppelin/rest/NotebookRestApiTest.java:236]: use `notebook` here.

**Review by Reamer — APPROVED** [review]
PullRequest LGTM, just a couple of really small comments.

File: zeppelin-server/src/main/java/org/apache/zeppelin/service/NotebookService.java (hunk 1 above)
Comment (huage1994) [reply zeppelin-server/src/main/java/org/apache/zeppelin/service/NotebookService.java:1015]: > Could you please put the following comment to this method. `notebook.getNoteByRevision(...) does not use the NoteCache, so we can return a Note object here.`

Sure! Thanks a lot for your review!

File: zeppelin-server/src/test/java/org/apache/zeppelin/rest/NotebookRestApiTest.java (hunk 2 above)
Comment (huage1994) [reply zeppelin-server/src/test/java/org/apache/zeppelin/rest/NotebookRestApiTest.java]: Thanks for giving me the good advice!  I've changed all the code like this in this PR.

File: zeppelin-server/src/test/java/org/apache/zeppelin/rest/NotebookRestApiTest.java (hunk 3 above)
Comment (huage1994) [reply zeppelin-server/src/test/java/org/apache/zeppelin/rest/NotebookRestApiTest.java]: Thanks for giving me the good advice!  I've changed all the code like this in this PR.

**Pull request merged**